src/adapters/DatabaseAdapter/index.ts

```diff
@@ -0,0 +1,92 @@
+import { firebaseApp } from "_/config/firabaseConfig";
+import {
+  getFirestore,
+  setDoc,
+  doc,
+  Firestore,
+  getDocs,
+  updateDoc,
+  deleteDoc,
+  getDoc,
+  collection,
+  query,
+  where,
+  orderBy,
+  CollectionReference,
+  DocumentData,
+  Query,
+  QuerySnapshot,
+} from "firebase/firestore";
+
+import { DatabaseType, ORDER, QueryOptions } from "./types";
+
+export class DatabaseAdapter implements DatabaseType {
```

Comment:
actually this is more a database repository

Reply:
Realizei a alteração